[BottomSheet] Create property to allow scrim color to be set on presentation controller.

danblakemore opened the pull request:
This new property behaves the same way that the analogous property on Dialogs does.
Also added a new test that tests the scrim color in the same way that the dialog
color is tested.

danblakemore committed:
Add Bottom Sheet dragon which helps reproduce an issue with bottom sheet centering and iPad Split View.

danblakemore committed:
Merge remote-tracking branch 'upstream/develop' into develop

danblakemore committed:
Merge remote-tracking branch 'upstream/develop' into develop

danblakemore committed:
Merge remote-tracking branch 'upstream/develop' into develop

danblakemore committed:
Updated copyright and import ordering.

danblakemore committed:
Merge remote-tracking branch 'upstream/develop' into develop

danblakemore committed:
Add a `scrimColor` property to the bottom sheet presentation controller.

This new property behaves the same way that the analogous property on Dialogs does.
Also added a new test that tests the scrim color in the same way that the dialog
color is tested.

jverkoey commented:
Ping @yarneo for review.

material-automation commented on components/BottomSheet/tests/unit/MDCBottomSheetCustomizationTests.m:
```diff
@@ -0,0 +1,68 @@
+  self.dummyContentViewController = [[UIViewController alloc] init];
+  self.bottomSheet =
+      [[MDCBottomSheetController alloc]
+          initWithContentViewController:self.dummyContentViewController];
```
clang-format suggested the following change:


```
-  self.bottomSheet =
-      [[MDCBottomSheetController alloc]
-          initWithContentViewController:self.dummyContentViewController];
+  self.bottomSheet = [[MDCBottomSheetController alloc]
+      initWithContentViewController:self.dummyContentViewController];
```

<!-- identifier: clang-format -->
<!-- File: components/BottomSheet/tests/unit/MDCBottomSheetCustomizationTests.m -->

material-automation commented:
clang-format recommended changes.

Consider installing [git-clang-format](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) and running the following command:

```bash
git clang-format $(git merge-base origin/develop HEAD)
```

<!-- identifier: clang-format -->

yarneo commented on components/BottomSheet/src/MDCBottomSheetPresentationController.h:
```diff
@@ -76,6 +76,13 @@
+/**
+ Customize the color of the background scrim.
+
+ Defaults to a semi-transparent Black.
```
can you please provide the exact color value

romoore replied:
Just a thought, but keeping it vague would allow us some latitude in the future if we need to shift the transparency by a few percent.  What do you think, @yarneo?

danblakemore replied:
I copied the comment [from MDCDialogPresentationController](https://github.com/material-components/material-components-ios/blob/develop/components/Dialogs/src/MDCDialogPresentationController.h#L66) and I kind of agree that specificity would be likely to get stale.

I'll update the PR to add the properties to the controller as well.

danblakemore replied:
Added. I also put a note in the test about a peculiarity/gotcha in the propagation of that new property.  Note that this is *not* how it's implemented for Dialogs (the transition controller isn't used at all) and that implementation doesn't have the same gotcha.  For Bottom Sheet, however, this matches the other properties that find their way from the sheet controller to the presentation controller, so it fits better with existing code.

I'm happy to change this to match Dialogs, if that's preferable, or leave it for cleanup later.  Either way, it will be in a separate PR.

material-automation commented:
## API diff detected the following changes

### BottomSheet

#### MDCBottomSheetPresentationController

*new* property: `scrimColor` in `MDCBottomSheetPresentationController`

#### MDCBottomSheetTransitionController(ScrimAccessibility)

*new* property: `scrimColor` in `MDCBottomSheetTransitionController(ScrimAccessibility)`

#### MDCBottomSheetController

*new* property: `scrimColor` in `MDCBottomSheetController`

<!-- identifier: apidiff -->

yarneo reviewed (changes requested):
Thanks for opening the PR Dan! Can you please expose this method in MDCBottomSheetController rather than only in the presentation controller. This means it should be exposed at MDCBottomSheetController and sent to MDCBottomSheetTransitionController which then sends it to MDCBottomSheetPresentationController.

Once that is done please add a unit test that tests the added MDCBottomSheetController api

danblakemore committed:
Add scrim color to the bottom sheet controller.

The scrim color is eventually set on the presentation controller
via the transition controller when the presentation controller is
first initialized.

yarneo commented on components/BottomSheet/src/MDCBottomSheetController.h:
```diff
@@ -50,6 +50,11 @@
 
+/**
+ The color applied to the sheet's background when presented by MDCBottomSheetPresentationController.
+ */
```
Can you please provide the Default here in the comment too

danblakemore committed:
Updating comment for scrim color on the controller.

danblakemore committed:
Merge remote-tracking branch 'upstream/develop' into develop

danblakemore committed:
Clang format

danblakemore commented:
I think this is all ready to go now.
